src/main/java/io/cryostat/net/web/http/api/beta/ProbeTemplateDeleteHandler.java

```diff
@@ -0,0 +1,144 @@
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.beta;
+
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.cryostat.core.agent.LocalProbeTemplateService;
+import io.cryostat.core.log.Logger;
+import io.cryostat.core.sys.FileSystem;
+import io.cryostat.messaging.notifications.NotificationFactory;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler;
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+
+public class ProbeTemplateDeleteHandler extends AbstractV2RequestHandler<Void> {
+
+    static final String PATH = "probes/:probetemplateName";
+
+    private static Logger logger;
+    private final NotificationFactory notificationFactory;
+    private final LocalProbeTemplateService probeTemplateService;
+    private final FileSystem fs;
+    private static final String NOTIFICATION_CATEGORY = "ProbeTemplateUploaded";
+
+    @Inject
+    ProbeTemplateDeleteHandler(
+            AuthManager auth,
+            NotificationFactory notificationFactory,
+            LocalProbeTemplateService probeTemplateService,
+            Logger logger,
+            FileSystem fs,
+            Gson gson) {
+        super(auth, gson);
+        this.notificationFactory = notificationFactory;
+        this.logger = logger;
+        this.probeTemplateService = probeTemplateService;
+        this.fs = fs;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.DELETE;
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return false;
```

Comment:
I think authentication should be required for this handler.

Reply:
Agreed, I had it marked as such for ease of testing, I've gone ahead and changed it now